Crash on any operation v1.0.1

I'll add more details to this issue, but strapped for time currently. First go at using this provider. If anyone picks this up, don't hesitate to reach out asking for more details. I provided what time allows.

This also happens with resource blocks, anything really. 
Happens on `plan` and `apply`

OpenTofu v1.9.0
Happens on both provider v0.5.1, v1.0.0 and v1.0.1

Provider block:
```
provider "cloudconnexa" {
  base_url = "https://REDACTED.openvpn.com/"
  client_id     = "REDACTED"
  client_secret = "REDACTED"
}
```

Simple data resource:
```
data "cloudconnexa_network" "REDACTED" {
  id = "REDACTED-UUID-WITH-DASHES"
}
```

```
Stack trace from the terraform-provider-cloudconnexa_v1.0.1 plugin:
  
  panic: runtime error: invalid memory address or nil pointer dereference
  [signal SIGSEGV: segmentation violation code=0x2 addr=0x38 pc=0x104dd74d4]
  
  goroutine 61 [running]:
  github.com/OpenVPN/terraform-provider-cloudconnexa/cloudconnexa.providerConfigure({0x105014ca2?, 0x6?}, 0x14000286280)
        github.com/OpenVPN/terraform-provider-cloudconnexa/cloudconnexa/provider.go:93 +0x124
  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0x1400027be30, {0x10530de70, 0x140004aef30}, 0x140004b8370)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.0/helper/schema/provider.go:391 +0x1d0
  github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0x140003d85d0, {0x10530de70?, 0x140004aeb40?}, 0x140004cc160)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.0/helper/schema/grpc_provider.go:647 +0x368
  github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0x14000304fa0, {0x10530de70?, 0x140004ae1e0?}, 0x140004b8140)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:587 +0x2ac
  github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x1052e2880, 0x14000304fa0}, {0x10530de70, 0x140004ae1e0}, 0x140004e2000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:557 +0x1c0
  google.golang.org/grpc.(*Server).processUnaryRPC(0x14000302200, {0x10530de70, 0x140004ae180}, 0x140004a40c0, 0x140003fe2d0, 0x10587fe70, 0x0)
        google.golang.org/grpc@v1.69.4/server.go:1392 +0xc44
  google.golang.org/grpc.(*Server).handleStream(0x14000302200, {0x10530e638, 0x140002f4000}, 0x140004a40c0)
        google.golang.org/grpc@v1.69.4/server.go:1802 +0x910
  google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.69.4/server.go:1030 +0x84
  created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
        google.golang.org/grpc@v1.69.4/server.go:1041 +0x13c
  
  Error: The terraform-provider-cloudconnexa_v1.0.1 plugin crashed!
  
  This is always indicative of a bug within the plugin. It would be immensely
  helpful if you could report the crash with the plugin's maintainers so that it
  can be fixed. The output above should help diagnose the issue.
  
  
  exit status 1
```

Thanks for the suggestions. I've removed trailing slash from `base_url` and have the same result. 

I will simplify my test case to creating a new resource in our sandbox WPC and will send trace-level logs if I can recreate. I will also see if I can recreate in a container-based delivery pipeline (basically try another environment).

More to follow tomorrow.
 
```
2025-04-10T14:06:30.226-0600 [DEBUG] created provider logger: level=debug
2025-04-10T14:06:30.226-0600 [INFO]  provider: configuring client automatic mTLS
2025-04-10T14:06:30.228-0600 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.opentofu.org/openvpn/cloudconnexa/1.0.1/darwin_arm64/terraform-provider-cloudconnexa_v1.0.1 args=[".terraform/providers/registry.opentofu.org/openvpn/cloudconnexa/1.0.1/darwin_arm64/terraform-provider-cloudconnexa_v1.0.1"]
2025-04-10T14:06:30.229-0600 [DEBUG] provider: plugin started: path=.terraform/providers/registry.opentofu.org/openvpn/cloudconnexa/1.0.1/darwin_arm64/terraform-provider-cloudconnexa_v1.0.1 pid=47314
2025-04-10T14:06:30.229-0600 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.opentofu.org/openvpn/cloudconnexa/1.0.1/darwin_arm64/terraform-provider-cloudconnexa_v1.0.1
2025-04-10T14:06:30.236-0600 [INFO]  provider.terraform-provider-cloudconnexa_v1.0.1: configuring server automatic mTLS: timestamp=2025-04-10T14:06:30.236-0600
2025-04-10T14:06:30.241-0600 [DEBUG] provider: using plugin: version=5
2025-04-10T14:06:30.241-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: plugin address: address=/var/folders/h1/nqfnm7wj5_z8lvxfq1hx007m0000gp/T/plugin3143349877 network=unix timestamp=2025-04-10T14:06:30.241-0600
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: panic: runtime error: invalid memory address or nil pointer dereference
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: [signal SIGSEGV: segmentation violation code=0x2 addr=0x38 pc=0x10285b4d4]
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: goroutine 14 [running]:
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: github.com/OpenVPN/terraform-provider-cloudconnexa/cloudconnexa.providerConfigure({0x102a98ca2?, 0x6?}, 0x14000386c80)
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	github.com/OpenVPN/terraform-provider-cloudconnexa/cloudconnexa/provider.go:93 +0x124
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0x14000269e30, {0x102d91e70, 0x140003677d0}, 0x140002591d0)
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.0/helper/schema/provider.go:391 +0x1d0
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0x140003405e8, {0x102d91e70?, 0x140003673e0?}, 0x1400007e5c0)
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.0/helper/schema/grpc_provider.go:647 +0x368
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0x14000271040, {0x102d91e70?, 0x14000366a80?}, 0x14000258fa0)
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:587 +0x2ac
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x102d66880, 0x14000271040}, {0x102d91e70, 0x14000366a80}, 0x14000386480, 0x0)
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:557 +0x1c0
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: google.golang.org/grpc.(*Server).processUnaryRPC(0x1400026e200, {0x102d91e70, 0x14000366a20}, 0x1400010e300, 0x140003662d0, 0x103303e70, 0x0)
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	google.golang.org/grpc@v1.69.4/server.go:1392 +0xc44
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: google.golang.org/grpc.(*Server).handleStream(0x1400026e200, {0x102d92638, 0x140003ec000}, 0x1400010e300)
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	google.golang.org/grpc@v1.69.4/server.go:1802 +0x910
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: google.golang.org/grpc.(*Server).serveStreams.func2.1()
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	google.golang.org/grpc@v1.69.4/server.go:1030 +0x84
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 20
2025-04-10T14:06:30.248-0600 [DEBUG] provider.terraform-provider-cloudconnexa_v1.0.1: 	google.golang.org/grpc@v1.69.4/server.go:1041 +0x13c
2025-04-10T14:06:30.249-0600 [ERROR] plugin.(*GRPCProvider).ConfigureProvider: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-04-10T14:06:30.249-0600 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.opentofu.org/openvpn/cloudconnexa/1.0.1/darwin_arm64/terraform-provider-cloudconnexa_v1.0.1 pid=47314 error="exit status 2"
2025-04-10T14:06:30.249-0600 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-04-10T14:06:30.249-0600 [ERROR] vertex "provider[\"registry.opentofu.org/openvpn/cloudconnexa\"]" error: Plugin did not respond
2025-04-10T14:06:30.249-0600 [INFO]  backend/local: plan operation completed
```

--- a/cloudconnexa/provider.go
+++ b/cloudconnexa/provider.go
@@ -3,6 +3,7 @@ package cloudconnexa
 import (
 	"context"
 	"fmt"
+
 	"github.com/openvpn/cloudconnexa-go-client/v2/cloudconnexa"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
@@ -117,15 +118,15 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 		baseUrl = "https://" + cloudId + ".api.openvpn.com"
 	}
 	cloudConnexaClient, err := cloudconnexa.NewClient(baseUrl, clientId, clientSecret)
-	cloudConnexaClient.UserAgent = fmt.Sprintf("terraform-provider-cloudconnexa/%v", version)
 	var diags diag.Diagnostics
 	if err != nil {
 		diags = append(diags, diag.Diagnostic{
 			Severity: diag.Error,
-			Summary:  "Unable to create client",
-			Detail:   fmt.Sprintf("Error: %v", err),
+			Summary:  "Unable to create CloudConnexa client",
+			Detail:   fmt.Sprintf("Failed to create CloudConnexa client with base URL '%s': %v", baseUrl, err),
 		})
 		return nil, diags
 	}
+	cloudConnexaClient.UserAgent = fmt.Sprintf("terraform-provider-cloudconnexa/%v", version)
 	return cloudConnexaClient, nil
 }

